### PR TITLE
[ISSUE #5923] Add example tiered storage backend service provider

### LINKED ISSUE #5923: [RIP-57] Add example tiered storage backend service provider
link to #5754 

As stated in the title, this issue aims to add a POSIX provider for tiered storage. It transfers message data from the default store directory to the mount point of storage medium for tiered storage. This provider is fully functional but for demonstration purposes only.

### PR DESCRIPTION
**Pull request by ShadowySpirits**
close #5923

### CODE REVIEW

**Comment by ShadowySpirits** [comment]
Please take a look. @aaron-ai @lollipopjin

File: tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/TieredStoreProvider.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import java.nio.ByteBuffer;
 import java.util.concurrent.CompletableFuture;
 
-public interface TieredStoreBackendProvider {
```
Comment (aaron-ai) [inline tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/TieredStoreProvider.java:22]: Could we use Java SPI mechanism to get the implementation?

File: tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/posix/PosixFileSegment.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+import org.apache.rocketmq.tieredstore.metrics.TieredStoreMetricsManager;
+import org.apache.rocketmq.tieredstore.provider.TieredFileSegment;
+import org.apache.rocketmq.tieredstore.util.TieredStoreUtil;
+
+import static org.apache.rocketmq.tieredstore.metrics.TieredStoreMetricsConstant.LABEL_FILE_TYPE;
+import static org.apache.rocketmq.tieredstore.metrics.TieredStoreMetricsConstant.LABEL_OPERATION;
+import static org.apache.rocketmq.tieredstore.metrics.TieredStoreMetricsConstant.LABEL_SUCCESS;
+import static org.apache.rocketmq.tieredstore.metrics.TieredStoreMetricsConstant.LABEL_TOPIC;
+
+public class PosixFileSegment extends TieredFileSegment {
```
Comment (aaron-ai) [inline tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/posix/PosixFileSegment.java:45]: According to readme, It seems that `PosixFileSegment` is just an example, there should be more comments to illustrate it.

File: tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/posix/PosixFileSegment.java [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        long baseOffset, TieredMessageStoreConfig storeConfig) {
+        super(fileType, messageQueue, baseOffset, storeConfig);
+
+        String basePath = storeConfig.getTieredStoreFilepath();
+        if (StringUtils.isBlank(basePath) || basePath.endsWith(File.separator)) {
+            this.basePath = basePath;
+        } else {
+            this.basePath = basePath + File.separator;
+        }
+        this.filepath = this.basePath
```
Comment (lollipopjin) [inline tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/posix/PosixFileSegment.java:68]: Take filePath length into condition？

File: tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/TieredStoreProvider.java (hunk 1 above)
Comment (ShadowySpirits) [reply tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/TieredStoreProvider.java:22]: The SPI mechanism is not applicable in this case. We want users to build their own provider by extending TieredFileSegment, which is an abstract class, not an interface. And tiered storage will create multiple file segment instances in multiple threads. It's more suitable to use the class name to initialize the file segment objects directly.

File: tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/posix/PosixFileSegment.java (hunk 3 above)
Comment (ShadowySpirits) [reply tieredstore/src/main/java/org/apache/rocketmq/tieredstore/provider/posix/PosixFileSegment.java:71]: The maximum length for a file name is 255 bytes on Linux. And it is 4096 bytes for the file path. This length is enough, no need to worry.

**Review by aaron-ai — APPROVED** [review]

**Review by lollipopjin — APPROVED** [review]
LGTM

**Pull request merged**